### ISSUE TITLE
Equivalent of Roslyn TypeSyntax / Get Types

### ISSUE THREAD

**Issue by ghost** [javaparser/javaparser#1470]
Hello,

I am trying to find the fastest way to resolve types.
With Microsoft roslyn all you need to collect are TypeSyntax nodes which reference other Classes, so basically everything inside your file whose origin is from another class in your project. Then you could just get the package name + class name of that Node.

What I used to do last year with JavaParser year was collect multiple nodes like MethodDeclarations, VariableDeclararators etc. and then use JavaParserFacader to solve them and get their FullyQualifiedName. This was rather complicated and not an optimal solution.

Since then there have a lot of changes to the API and I am seeing that JavaParserFacade will soon be deprecated and I do not necessarily find ParserConfiguration/JavaParser intuitive.

TL;DR:
Best way to solve Types?
In a best case scenario I would just like to collect Nodes with cu.findall(Node.class) and then
run a method that returns their types/FullyQualifiedNames.

**Comment by ghost** [comment]
I should be able to figure this out and most likely will write a guide about it, as no documentation about the new API exists as of right now.

Could I just get clarification which classes exactly are deprecated? 
Am I correct in assuming that `CombinedTypeSolver`, [`JavaParserFacade`](https://github.com/javaparser/javaparser/issues/1377) and `JavaParser `(`SourceRoot.getJavaParser()` suggests so) will be removed in the future?

 So basically we should now be using SourceRoot and ParserConfiguration for everything

Edit: What about all the other [JavaParser*](https://github.com/javaparser/javaparser/issues/1395) classes?

**Comment by matozoid** [comment]
@ftomassetti?

**Comment by ghost** [comment]
Oh well, pretty much finished now anyway.

I'll be posting the guide here towards the weekend and then leave this issue up for another week or two before I close it, so other people can see it as well.

**Comment by ghost** [comment]
It's done https://sdoplace.wordpress.com/2018/03/24/english-how-to-parse-a-project-and-get-types-of-nodes-with-javaparser/ . 

If there are graves mistakes or things I missed/that are important please let me know here or on the site.  When I have more time to play with Javaparser, I'll most likely update the guide with new information.

Edit: If you don't necessarily want to leave this issue open for another week or so, feel free to close it. it's really no problem for me :)

**Comment by matozoid** [comment]
Cool, I'll refer people to that if they have questions :-)

Of course, the state of the documentation is horrible right now: we made a huge change and then we found we had no time to update the docs. So, how would you feel about having your name on the JavaParser book too? You only need to update the symbol solver chapter ;-)

**Comment by ghost** [comment]
Haha, I feel like my knowledge is by far not good enough to tackle that right now

Also I just found these [Presentation Slides](https://tomassetti.me/wp-content/uploads/2017/12/JavaParser-JUG-Milano.pdf) which actually are pretty much up to date. Wish I had found them earlier. Should be the best ressource as of right now
